client/coral-plugin-commentbox/CommentBox.js

```diff
@@ -68,6 +101,15 @@ class CommentBox extends Component {
     this.setState({body: ''});
   }
 
+  addCommentHooks = (hooks = {}) => {
+    if (typeof hooks === 'object') {
```

Comment:
That implementation triggers a lot of issues when a user tries to add the same hook again. I won't move forward with it.

I will replace the API for registerHook to be similar to unregister.